java/kafka/deployment-oauth.yaml

```diff
@@ -69,16 +69,20 @@ spec:
       - name: java-kafka-producer
         image: quay.io/strimzi-examples/java-kafka-producer:latest
         env:
-          - name: BOOTSTRAP_SERVERS
+          - name: KAFKA_BOOTSTRAP_SERVERS
             value: my-cluster-kafka-bootstrap:9092
-          - name: TOPIC
+          - name: STRIMZI_TOPIC
             value: my-topic
-          - name: DELAY_MS
+          - name: STRIMZI_DELAY_MS
             value: "1000"
-          - name: LOG_LEVEL
+          - name: STRIMZI_LOG_LEVEL
             value: "INFO"
-          - name: MESSAGE_COUNT
+          - name: STRIMZI_MESSAGE_COUNT
             value: "1000000"
+          - name: KAFKA_KEY_SERIALIZER
+            value: "org.apache.kafka.common.serialization.StringSerializer"
+          - name: KAFKA_VALUE_SERIALIZER
+            value: "org.apache.kafka.common.serialization.StringSerializer"
```

Comment:
Is this really enough? Did you tried it? My guess is that both here (but also in the consumer), you are missing at least:
* Security protocol which needs to be set to PLAINTEXT_SASL or something like that
* SASL mechanism configuration
* The JAAS configuration
* Callback handler configuration

Reply:
Tested all of the above including SSL and  SASL_PLAINTEXT for Security Protocol.